File: packages/zynth-async-storage/ios/ZynthAsyncStorage.swift
//
//  ZynthAsyncStorage.swift
//  ZynthAsyncStorage
//
//  Public interface for ZynthAsyncStorage module
//

import Foundation
import ZynthKit

@objc(ZynthAsyncStorage)
public class ZynthAsyncStorage: NSObject {
  private static var moduleInstance: ZynthAsyncStorageModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard moduleInstance == nil else {
      print("[ZynthAsyncStorage] Module already initialized")
      return
    }

    let module = ZynthAsyncStorageModule()
    runtime.installModules([module])
    moduleInstance = module
    print("[ZynthAsyncStorage] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}

File: packages/zynth-async-storage/android/ZynthAsyncStorage/src/main/java/dev/zynth/asyncstorage/ZynthAsyncStorage.kt
package dev.zynth.asyncstorage

import android.app.Activity
import com.zynth.kit.runtime.ZynthRuntime

/**
 * Public interface for ZynthAsyncStorage module.
 */
object ZynthAsyncStorage {
    private var moduleInstance: ZynthAsyncStorageModule? = null

    /**
     * Initialize the AsyncStorage module with an Activity and ZynthRuntime instance.
     */
    @JvmStatic
    fun initialize(activity: Activity, runtime: ZynthRuntime) {
        if (moduleInstance != null) {
            android.util.Log.w("ZynthAsyncStorage", "Module already initialized")
            return
        }

        moduleInstance = ZynthAsyncStorageModule(activity.applicationContext)
        runtime.installModules(listOf(moduleInstance!!))
        android.util.Log.d("ZynthAsyncStorage", "Module initialized")
    }

    /**
     * Clean up the module (called during app teardown).
     */
    @JvmStatic
    fun cleanup() {
        moduleInstance = null
    }
}
